### DISCUSSION TITLE
Use two different storage class for primary and replicas

### DISCUSSION THREAD

**Discussion by sylvainmetayer** [cloudnative-pg/cloudnative-pg#5511] [Q&A] [answered]
Hi,

I have two storage class on my cluster, one dedicated to each region of my k8s cluster (using `allowedTopologies` to my storage class to ensure it is properly scheduled)

I want to create a cluster with a primary instance using the storage class A, and the replicas using a storage B.

Is my only option to create 2 clusters (Following the [Replica cluster via streaming sample](https://cloudnative-pg.io/documentation/current/samples/)) or did I miss another option to customize the storage class for the replicas ?

**Comment by gbartolini** [comment, marked as answer]
Hello,

Thanks for using CNPG.

> I have two storage class on my cluster, one dedicated to each region of my k8s cluster (using `allowedTopologies` to my storage class to ensure it is properly scheduled)

Sure.

> I want to create a cluster with a primary instance using the storage class A, and the replicas using a storage B.

That is not possible within the same HA cluster. Per definition, all instances are equal in a HA cluster as any of them can become primary (remember cattle vs pets?). Also, what happens if a replica is promoted? Should we move the storage in the other storage class?

Maybe in the future, once we have done all the important features that are still missing, we can think of service classes within a cluster, but it is not coming soon (unless we find a customer that needs this feature so badly).

> Is my only option to create 2 clusters (Following the [Replica cluster via streaming sample](https://cloudnative-pg.io/documentation/current/samples/)) or did I miss another option to customize the storage class for the replicas ?

Exactly. Based on your definition, the standalone replica cluster is the way to go.

I hope this helps.

Ciao,
Gabriele
